packages/api/src/Base.ts

```diff
@@ -300,8 +300,8 @@ export default abstract class ApiBase<OnCall> implements ApiBaseInterface<OnCall
       this._runtimeVersion = await this._rpcBase.chain.getRuntimeVersion();
       this._genesisHash = await this._rpcBase.chain.getBlockHash(0);
 
-      const extrinsics = extrinsicsFromMeta(this.runtimeMetadata);
-      const storage = storageFromMeta(this.runtimeMetadata);
+      const extrinsics = extrinsicsFromMeta(this.runtimeMetadata.asV0);
```

Comment:
Question: if i understand correctly, each time we call .asV0, it will recalculate v1ToV0 function (a bit heavy), which might not be super optimized.

Though if v0 is going away soon, I prefer not to forget about this optimization.

Reply:
There is a cache for the v1 -> v0 conversion - https://github.com/polkadot-js/api/pull/632/files#diff-a208134f9b85707425c723f3705ebb29R86

Reply:
I think this type of "convert once" thing will probably follow-around each time we have a bump.

Reply:
Ah correct